python/ray/serve/_private/deployment_state.py

```diff
@@ -1593,6 +1607,45 @@ def deploy(self, deployment_info: DeploymentInfo) -> bool:
         self._backoff_time_s = 1
         return True
 
+    def drop_invalid_handle_metrics(self, alive_serve_actor_ids: Set[str]) -> None:
+        """Drops handle metrics that are no longer valid.
+
+        This includes handles that live on Serve Proxy or replica actors
+        that have died and handles from which the controller hasn't
+        received an update for too long.
+        """
+
+        timeout_s = max(
+            2 * self.autoscaling_policy_manager.get_metrics_interval_s(),
+            RAY_SERVE_MIN_HANDLE_METRICS_TIMEOUT_S,
+        )
+        for handle_id, handle_metric in list(self.handle_requests.items()):
+            # Drop metrics for handles that are on Serve proxy/replica
+            # actors that have died
+            if (
+                handle_metric.handle_component
+                in [HandleComponent.PROXY, HandleComponent.REPLICA]
```

Comment:
could make this a property on `HandleSource` like `.is_serve_component() -> bool`

Reply:
not sure if this is actually better or not -- up to you

Reply:
I like it, changed!